File: C/waterbottles.c
#include <stdio.h>
int numWaterBottles(int numBottles, int numExchange) {
    int total = numBottles;
    int empty = numBottles;

    while (empty >= numExchange) {
        int newBottles = empty / numExchange;
        total += newBottles;
        empty = newBottles + (empty % numExchange);
    }

    return total;
}

int main() {
    printf("%d\n", numWaterBottles(9, 3));
    return 0;
}

File: Python/Waterbottles.py
def numWaterBottles(numBottles, numExchange):
    total = numBottles
    empty = numBottles

    while empty >= numExchange:
        new_bottles = empty // numExchange   # exchange empty bottles for new full ones
        total += new_bottles                 # drink those new bottles
        empty = new_bottles + (empty % numExchange)  # new empties + leftovers

    return total


# Example
print(numWaterBottles(9, 3))  # Output: 13
